型エラー修正: 抽象メソッド実装とprivateプロパティの問題

## 概要
Issue #227で対応した型チェックエラー修正の続きです。
アーキテクチャレベルの検討が必要な型エラーが残っているため、個別に対応します。

## 対応が必要なエラー

### 1. Enemy.ts: 抽象メソッド実装漏れ
```
error TS2654: Non-abstract class 'Enemy' is missing implementations for the following members of 'Entity': 'getAnimationDefinitions', 'getPaletteDefinition'.
```

### 2. Spider.ts/EnergyBullet.ts: privateプロパティの問題
```
error TS2415: Class 'Spider' incorrectly extends base class 'Enemy'.
  Property 'stateTimer' is private in type 'Spider' but not in type 'Enemy'.

error TS2415: Class 'EnergyBullet' incorrectly extends base class 'Entity'.
  Property 'animationTime' is private in type 'EnergyBullet' but not in type 'Entity'.
```

### 3. PhysicsSystem.ts: PhysicsEntity vs Entityの型互換性
```
error TS2430: Interface 'PhysicsEntity' incorrectly extends interface 'Entity'.
  Property 'airResistance' is optional in type 'PhysicsEntity' but required in type 'Entity'.
```

## 対応方針

1. **抽象メソッド実装**: Enemyクラスに必要なメソッドを実装するか、抽象クラスとして定義
2. **privateプロパティ**: protectedに変更するか、基底クラスで定義
3. **型互換性**: airResistanceプロパティの必須/オプション統一

## 関連Issue
- #227: 型の不一致エラー修正（TS2345, TS2322）
- #224: TypeScript型チェックエラーの解消

--- a/src/entities/Enemy.ts
+++ b/src/entities/Enemy.ts
@@ -4,6 +4,7 @@ import { Player } from './Player';
 import { PixelRenderer } from '../rendering/PixelRenderer';
 import { Logger } from '../utils/Logger';
 import type { BaseEntityConfig } from '../config/ResourceConfig';
+import type { AnimationDefinition, EntityPaletteDefinition } from '../types/animationTypes';
 export type EnemyState = 'idle' | 'walk' | 'hurt' | 'dead';
 
 /**
@@ -229,5 +230,37 @@ export class Enemy extends Entity {
         
         const hpWidth = (this.health / this.maxHealth) * barWidth;
         renderer.drawRect(barX, barY, hpWidth, barHeight, 0x31);
+    }
+    
+    /**
+     * Get animation definitions for enemy
+     * Override in derived classes for specific enemy animations
+     */
+    protected getAnimationDefinitions(): AnimationDefinition[] {
+        return [
+            {
+                id: 'idle',
+                sprites: ['enemies/generic_enemy.json'],
+                frameDuration: 0,
+                loop: false
+            }
+        ];
+    }
+    
+    /**
+     * Get palette definition for enemy
+     * Override in derived classes for specific enemy palettes
+     */
+    protected getPaletteDefinition(): EntityPaletteDefinition {
+        return {
+            default: {
+                colors: [
+                    null,
+                    0x00,
+                    0x10,
+                    0x30
+                ]
+            }
+        };
     }
 }

--- a/src/entities/Enemy.ts
+++ b/src/entities/Enemy.ts
@@ -237,14 +237,7 @@ export class Enemy extends Entity {
      * Override in derived classes for specific enemy animations
      */
     protected getAnimationDefinitions(): AnimationDefinition[] {
-        return [
-            {
-                id: 'idle',
-                sprites: ['enemies/generic_enemy.json'],
-                frameDuration: 0,
-                loop: false
-            }
-        ];
+        return [];
     }
     
     /**

--- a/src/entities/enemies/Spider.ts
+++ b/src/entities/enemies/Spider.ts
@@ -27,7 +27,7 @@ export class Spider extends Enemy implements EntityInitializer {
     private initialY: number;
     private threadY: number;
     private waitTime: number;
-    private stateTimer: number;
+    declare stateTimer: number;
     private patrolRange: number;
     private lastPlayerCheck: number;
     private detectionCooldown: number;

--- a/src/entities/projectiles/EnergyBullet.ts
+++ b/src/entities/projectiles/EnergyBullet.ts
@@ -15,7 +15,7 @@ export class EnergyBullet extends Entity implements EntityInitializer {
     private lifeTime: number;
     private maxLifeTime: number;
     private damage: number;
-    private animationTime: number;
+    declare animationTime: number;
     private destroyed: boolean = false;
     private originX: number;
     private originY: number;

--- a/src/physics/PhysicsSystem.ts
+++ b/src/physics/PhysicsSystem.ts
@@ -46,10 +46,10 @@ type CollisionMatrix = {
 };
 
 interface PhysicsEntity extends Entity {
-    physicsLayer?: PhysicsLayer;
-    airResistance?: number;
-    gravityScale?: number;
-    maxFallSpeed?: number;
+    physicsLayer: PhysicsLayer;
+    airResistance: number;
+    gravityScale: number;
+    maxFallSpeed: number;
 }
 
 interface IRenderer {

